### PR TITLE
meal and reservation endpoints

### PR DESCRIPTION
**Pull request by hamoudi-hamoudi**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -1,15 +1,91 @@
...
+      await knex("meal").insert(addMeal);
+      res.status(201).json(`meal ${addMeal.title} added successfully`);
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  }
+});
+router.get("/:id", async (req, res) => {
+  const id = req.params.id;
+  if (isNaN(id)) {
```
Comment (raihanmehran) [inline src/backend/api/meals.js]: also add a check for if 'id' exists and also add it to the rest of endpoints which receives `id` from `req.params`

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -1,15 +1,91 @@
...
+  } else {
+    try {
+      updatedMeal.id = id;
+      const meal = await knex("meal").where({ id });
+      meal.length === 0
+        ? res.status(200).json(`meal not found`)
+        : (await knex("meal").where({ id }).update(updatedMeal)) &&
+          res
+            .status(200)
+            .json(`meal ${updatedMeal.title} updated successfully`);
```
Comment (raihanmehran) [inline src/backend/api/meals.js]: Well, this is so nice that you understand usage of ternary operators. But what I would suggest in this, it is more understandable and simple to use if conditions.
Your refactored code will look like this:
`if (meal.length === 0) res.status(200).json('meal not found');`
`await knex("meal").where({ id }).update(updatedMeal);`
`res.status(200).json('Meal ${updatedMeal.title} updated successfully');`

Comment (raihanmehran) [reply src/backend/api/meals.js]: and if you consider it for the rest of the endpoints, it will make code more readable and easier to understand

**Review by raihanmehran — COMMENTED** [review]
At last, nice job @hamoudi-hamoudi 👍 

Happy Coding :)

File: src/backend/api/meals.js (hunk 1 above)
Comment (hamoudi-hamoudi) [reply src/backend/api/meals.js]: how do i make this check ? Error: ENOENT: no such file or directory, stat 'C:\Users\ba3ba3\Desktop\meal-sharing\dist\index.html' 
I keep having this error , also tried req.body.myParam

**Pull request merged**